### ISSUE TITLE
Config option to disable SMIME verification while keeping PGP verification

### ISSUE THREAD

**Issue by sebastianw** [neomutt/neomutt#3068]
## What would you like to be able to do?

Right now there is only one way to prevent neomutt from trying to verify S/MIME signatures when using gpgme.
You need to set `set crypt_verify_sig=no` in the configuration. But this also disables PGP signature verification.

## What needs to change in NeoMutt?

There should be separate options to disable S/MIME (or PGP for that matter) signature verification, like this:

```
set smime_verify_sig=no
set pgp_verify_sig=no
```

## Have you tried any alternative methods?

Tried to unset all S/MIME commands:
```
set smime_verify_command=""
set smime_verify_opaque_command=""
set smime_decrypt_command=""
```
Does nothing when gpgme is used.

Tried to override the mime-type of the S/MIME certificate:

```
mime_lookup application/x-pkcs7-certificates text/plain
```

Signature verification still happens.

## Additional Info

S/MIME verification slows neomutt down a lot when viewing a message with a S/MIME signature. Also there is a dialog popup asking if I want to trust the issuer which breaks the whole screen rendering. I don't need and don't want S/MIME but I want to keep PGP.

**Comment by charles2910** [comment]
Hey there, there is a bug filled in Debian's BTS ([#988127 - neomutt hangs for minutes while checking S/MIME signed mails](https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=988127)) that is related to this issue, particularly with the slow down caused by S/MIME verification.

Daniel Gröber did a [thorough investigation on the matter](https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=988127#79) and came up with these findings:

> Some things I found: since I set crypt_use_gpgme=yes gpgme apparently
> handles S/MIME directly (didn't know gpg supported it) and the
> "backend" is /usr/bin/gpgsm.
> 
> So a very nasty hack is to get rid of this issue is to just symlink
> gpgsm to /usr/bin/false somewhere on your $PATH:
> 
>     # ln -s /usr/bin/false gpgsm
> 
> Looking at the code I found the original sin to be at
> ncrypt/cryptglue.c:crypt_init:
> 
>     #ifdef CRYPT_BACKEND_GPGME
>       if (c_crypt_use_gpgme)
>       {
>         crypto_module_register(&CryptModPgpGpgme);
>         crypto_module_register(&CryptModSmimeGpgme);
>       }
>     #endif
> 
> this makes it so crypt_use_gpgme=yes enables both gpg and smime
> support with no way to disable smime at init or message verification
> time. Not even hooks will help since the crypt module registration
> runs only once.
> 
> IMO this is unacceptable as I have no interest in being exposed to the
> vulnerability surface area of smime despite not having any use for it,
> so I'm planning to propose a patch to neomutt to move the smime
> registration to a seperate rc variable.

I'm not sure if I should open another issue to discuss it or we can use this one, but I think the request to enable the backends separately is reasonable (although I'm far from an expert here).

**Comment by dave-white** [comment]
In the patch linked just above, I took a first pass at splitting the enabling of PGP and S/MIME signature verification into two independent configuration variable, plus adding message flags and logic to set those flags as appropriate.  Should this lead to any appetite for discussing the issue further or for going ahead and implementing a solution, I'd be happy to continue the work and get advice on how to refine what I've got.
